Fabric.Terminology.SqlServer/Persistence/SqlClientTermValueSetRepository.UoWOperation.cs

```diff
@@ -0,0 +1,138 @@
+                .Select(dto => new Operation
+                {
+                    Value = dto,
+                    OperationType = OperationType.Delete
+                }).ToList();
+        }
+
+        private IReadOnlyCollection<Operation> BuildCodeCountOperationList(
+            Guid valueSetGuid,
+            IEnumerable<ValueSetCodeDto> allCodeDtos)
+        {
+            var originalCounts = this.uowManager.GetCodeCountDtos(valueSetGuid);
+            var existingCodeSystems = originalCounts.Select(c => c.CodeSystemGUID);
+
+            var allCodes = allCodeDtos as ValueSetCodeDto[] ?? allCodeDtos.ToArray();
+            var allCodeSystems = allCodes.Select(c => c.CodeSystemGuid).Distinct();
+
+            var recounts = allCodeSystems.Select(
+                            codeSystemGuid => new
+                            {
+                                codeSystemGuid,
+                                valueSetDto = allCodes.First(dto => dto.CodeSystemGuid == codeSystemGuid)
+                            })
+                            .Select(o =>
+                            new ValueSetCodeCountDto(o.valueSetDto.ValueSetGUID,
+                                o.valueSetDto.CodeSystemGuid,
+                                o.valueSetDto.CodeSystemNM,
+                                allCodes.Count(c => c.CodeSystemGuid == o.codeSystemGuid)))
+                                .ToList();
+
+            var operations = recounts.Select(nc =>
+                Maybe.From(originalCounts.FirstOrDefault(ec => ec.CodeSystemGUID == nc.CodeSystemGUID))
+                .Select(
+                    dto =>
+                    {
+                        var op = new Operation();
+                        if (dto.CodeSystemPerValueSetNBR != nc.CodeSystemPerValueSetNBR)
+                        {
+                            dto.CodeSystemPerValueSetNBR = nc.CodeSystemPerValueSetNBR;
+                            op.OperationType = OperationType.Update;
+                        }
+                        else
+                        {
+                            op.OperationType = OperationType.None;
+                        }
+
+                        op.Value = dto;
+                        return op;
+                    })
+                .Else(() => new Operation
+                {
+                    Value = nc
+                })).ToList();
+
+            // finally ensure that any existing counts that were not in the new counts are removed
+            // e.g. all codes from a particular code system were removed
+            var removers = originalCounts.Where(oc => !existingCodeSystems.Contains(oc.CodeSystemGUID));
+
+            operations.AddRange(removers.Select(r => new Operation
+            {
+                Value = r,
+                OperationType = OperationType.Delete
+            }));
+
+            return operations.Where(op => op.OperationType != OperationType.None).ToList();
```

Comment:
Why are we adding "none" operations instead of just not including those operations in the first place? 
Leveraging `Maybe` as you do currently, you could use `SelectMany` and `Where` to filter out the items that have the same value, and then you wouldn't need the `Else`.

In a foreach, you can just use an `if` statement to skip adding the operation to the list.